neurobooth_os/iout/lsl_streamer.py

```diff
@@ -1,163 +1,291 @@
+
+
+def start_mouse_stream(_, **device_args):
+    from neurobooth_os.iout.mouse_tracker import MouseStream
+    device = MouseStream(**device_args)
+    device.start()
+    return device
+
+
+def start_mbient_stream(_, **device_args):
+    from neurobooth_os.iout.mbient import Mbient
+    device = Mbient(**device_args)
+    if not device.prepare():
+        return None
+    device.start()
+    return device
+
+
+def start_mbient_mock_stream(_, **device_args):
+    from neurobooth_os.mock.mock_device_streamer import MockMbient
+    device = MockMbient(**device_args)
+    device.start()
+    return device
+
+
+def start_intel_stream(_, **device_args):
+    from neurobooth_os.iout.camera_intel import VidRec_Intel
+    return VidRec_Intel(**device_args)
+
+
+def start_intel_mock_stream(_, **device_args):
+    from neurobooth_os.mock.mock_device_streamer import MockCamera
+    return MockCamera(**device_args)
+
+
+def start_flir_stream(_, **device_args):
+    from neurobooth_os.iout.flir_cam import VidRec_Flir
+    return VidRec_Flir(**device_args)
+
+
+def start_iphone_stream(_, **device_args):
+    from neurobooth_os.iout.iphone import IPhone
+    device = IPhone(name="IPhoneFrameIndex", **device_args)
+    return device if device.prepare() else None
+
+
+def start_yeti_stream(_, **device_args):
+    from neurobooth_os.iout.microphone import MicStream
+    device = MicStream(**device_args)
+    device.start()
+    return device
+
+
+# --------------------------------------------------------------------------------
+# Device Configurations
+# TODO: Server assignment and device setup should all be derived from config files!!!
+# --------------------------------------------------------------------------------
+
+SERVER_ASSIGNMENTS: Dict[str, List[str]] = {
+    'acquisition': [
+        'Intel_D455_1', 'Intel_D455_2', 'Intel_D455_3', 'FLIR_blackfly_1', 'IPhone_dev_1',
+        'Mbient_BK_1', 'Mbient_LH_1', 'Mbient_LH_2', 'Mbient_RH_1', 'Mbient_RH_2',
+        'Mic_Yeti_dev_1',
+    ],
+    'presentation': ['Eyelink_1', 'Mouse', 'Mbient_LF_1', 'Mbient_LF_2', 'Mbient_RF_2'],
+    'dummy_acq': ['mock_Intel_1', 'mock_Mbient_1', 'mock_Mbient_2'],
+    'dummy_stm': [],
+}
+
+
+DEVICE_START_FUNCS: Dict[str, Callable] = {
+    'Eyelink_1': start_eyelink_stream,
+    'FLIR_blackfly_1': start_flir_stream,
+    'Intel_D455_1': start_intel_stream,
+    'Intel_D455_2': start_intel_stream,
+    'Intel_D455_3': start_intel_stream,
+    'IPhone_dev_1': start_iphone_stream,
+    'Mbient_BK_1': start_mbient_stream,
+    'Mbient_LF_1': start_mbient_stream,
+    'Mbient_LF_2': start_mbient_stream,
+    'Mbient_LH_1': start_mbient_stream,
+    'Mbient_LH_2': start_mbient_stream,
+    'Mbient_RF_2': start_mbient_stream,
+    'Mbient_RH_1': start_mbient_stream,
+    'Mbient_RH_2': start_mbient_stream,
+    'Mic_Yeti_dev_1': start_yeti_stream,
+    'Mouse': start_mouse_stream,
+    # ------------------------------
+    'mock_Intel_1': start_intel_mock_stream,
```

Comment:
In the future we should probably read this dictionary from a file so that we aren't hard-coding mocks in the production code.

Reply:
Hard agree - I wanted to arrange all the hardcoded stuff in a way that should make it easy to replace with file-based configs.